apps/backoffice-v2/src/pages/Entities/components/CaseGeneration/context/case-generation-context/hocs/withCaseGeneration/withCaseGeneration.tsx

```diff
@@ -0,0 +1,17 @@
+import { CaseGenerationContextProvider } from '@/pages/Entities/components/CaseGeneration/context/case-generation-context/Provider';
+
+export function withCaseGeneration<TComponentProps extends object>(
```

Comment:
Why do we need an HoC for a context provider or at least for the case generation?

Reply:
HoC is needed to avoid generating additional component layers around CaseCreation so it could have access to context. Usage is here `src/pages/Entities/components/CaseCreation/CaseCreation.tsx`